src/PushThread.cpp

```diff
@@ -982,6 +982,13 @@ void  PushThreadShared::remove_attr(string &signame)
 	push_back_cmd(cmd);
 }
 
+void PushThreadShared::add_attr(string &signame, int data_type, int data_format, int write_type)
+{
+	//------Configure DB------------------------------------------------
+	HdbCmdData *cmd = new HdbCmdData(DB_ADD, data_type, data_format, write_type, signame);
```

Comment:
I get this error when compiling:

```
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp: In member function ‘void HdbEventSubscriber_ns::PushThreadShared::add_attr(std::__cxx11::string&, int, int, int)’:
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp:988:35: error: ‘DB_ADD’ was not declared in this scope
  HdbCmdData *cmd = new HdbCmdData(DB_ADD, data_type, data_format, write_type, signame);
                                   ^~~~~~
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp:988:35: note: suggested alternative: ‘T_ADD’
  HdbCmdData *cmd = new HdbCmdData(DB_ADD, data_type, data_format, write_type, signame);
                                   ^~~~~~
                                   T_ADD
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp: In member function ‘virtual void* HdbEventSubscriber_ns::PushThread::run_undetached(void*)’:
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp:1148:10: error: ‘DB_ADD’ was not declared in this scope
     case DB_ADD:
          ^~~~~~
/home/reynald/GIT/tango-controls-hdbpp/hdbpp-es/src/PushThread.cpp:1148:10: note: suggested alternative: ‘T_ADD’
     case DB_ADD:
          ^~~~~~
          T_ADD

```

I compiled using the following CMake command:
`cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_VERBOSE_MAKEFILE=True -DCMAKE_INSTALL_PREFIX=/home/reynald/install/hddbpp  -DFETCH_LIBHDBPP=ON -DFETCH_LIBHDBPP_TAG=exp-refactor`

It looks like I used a wrong version of libhdbpp (exp-refactor)...? I tried with master branch and CMake command was returning other errors.

It looks like this DB_ADD constant should be defined in hdb++/AbstractDB.h file?

Reply:
yes I made a commit on libhdbpp-timescale, but not on libhdbpp as I was bypassing it, my bad. If you want to compile it just get the AbstractDB.h from libhdbpp-timescale, I'll make a commit on libhdbpp